bin/update_folders.py

```diff
@@ -105,6 +124,21 @@ def update_folders(
             f"Non-empty prediction and log files must be provided to update folders for submission {submission_id}. Exiting."
         )
 
+    # Make a record of the file entity's Synapse ID so it can be stored as an annotation for the given submission
+    file_synid = file_entity.id
+    output_annotation = {f"{folder_name}_id": file_synid}
+    output_annotation_filename = f"output_annotation_{submission_id}.json"
+    print(f"New annotation will be added for submission: {submission_id}")
+    print(f"Synapse ID for {folder_name} is {file_synid}")
+
+    # Read existing data if file exists, otherwise `existing_annotation` is an empty dictionary
+    existing_annotation = load_data(output_annotation_filename)
```

Comment:
Sorry if I missed this somewhere, but under what circumstances would this file already exist?

Reply:
`update_folders.py` gets executed twice in `update_folders.nf`: once for the predictions file (if it's not an `INVALID` file), and once for the docker log. This logic accounts for the execution for the docker log.

Reply:
Ah right, thanks!